src/args/argbuilder/option.rs

```diff
@@ -30,17 +26,131 @@ pub struct OptBuilder<'n> {
     pub min_vals: Option<u8>,
     pub max_vals: Option<u8>,
     pub val_names: Option<BTreeSet<&'n str>>,
-    pub empty_vals: bool,
-    pub global: bool,
     pub validator: Option<Rc<Fn(String) -> StdResult<(), String>>>,
     /// A list of names for other arguments that *mutually override* this flag
     pub overrides: Option<Vec<&'n str>>,
-    pub hidden: bool
+    pub settings: ArgFlags,
+}
+
+impl<'n> OptBuilder<'n> {
+    pub fn new(name: &'n str) -> Self {
+        OptBuilder {
+            name: name,
+            short: None,
+            long: None,
+            help: None,
+            blacklist: None,
+            possible_vals: None,
+            requires: None,
+            num_vals: None,
+            min_vals: None,
+            max_vals: None,
+            val_names: None,
+            validator: None,
+            overrides: None,
+            settings: ArgFlags::new()
+        }
+    }
+
+    pub fn from_arg(a: &Arg<'n, 'n, 'n, 'n, 'n,'n>,
+                    reqs: &mut Vec<&'n str>) -> Self {
+        if a.short.is_none() && a.long.is_none() {
+            panic!("Argument \"{}\" has take_value(true), yet neither a short() or long() \
+                were supplied", a.name);
+        }
+        // No need to check for .index() as that is handled above
+        let mut ob = OptBuilder {
+            name: a.name,
+            short: a.short,
+            long: a.long,
+            blacklist: None,
+            help: a.help,
+            possible_vals: None,
+            num_vals: a.num_vals,
+            min_vals: a.min_vals,
+            max_vals: a.max_vals,
+            val_names: a.val_names.clone(),
+            requires: None,
+            validator: None,
+            overrides: None,
+            settings: ArgFlags::new()
+        };
+        if a.multiple {
+            ob.settings.set(&ArgSettings::Multiple);
+        }
+        if a.required {
+            ob.settings.set(&ArgSettings::Required);
+        }
+        if a.global {
+            ob.settings.set(&ArgSettings::Global);
+        }
+        if !a.empty_vals {
+            ob.settings.unset(&ArgSettings::Global);
+        }
+        if a.hidden {
+            ob.settings.set(&ArgSettings::Hidden);
+        }
+        if let Some(ref vec) = ob.val_names {
+            ob.num_vals = Some(vec.len() as u8);
+        }
+        if ob.min_vals.is_some() && !a.multiple {
+            panic!("Argument \"{}\" does not allow multiple values, yet it is expecting {} \
+                values", ob.name, ob.num_vals.unwrap());
```

Comment:
Good catch, those should come out entirely because it's a legacy `panic!` from old features